pgpainless_signer/src/main/java/org/eclipse/packager/rpm/signature/pgpainless/PGPainlessSignatureProcessor.java

```diff
@@ -0,0 +1,122 @@
+import org.pgpainless.encryption_signing.SigningOptions;
+import org.pgpainless.key.protection.SecretKeyRingProtector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+
+public class PGPainlessSignatureProcessor implements SignatureProcessor {
+
+    protected final EncryptionStream signingStream;
+    private final Logger logger = LoggerFactory.getLogger(PGPainlessSignatureProcessor.class);
+
+    public PGPainlessSignatureProcessor(PGPSecretKeyRing key, SecretKeyRingProtector keyProtector, int hashAlgorithm) {
+        OutputStream sink = new OutputStream() {
+            @Override
+            public void write(int i) {
+                // Discard plaintext
+            }
+        };
+        try {
+            signingStream = PGPainless.encryptAndOrSign()
+                .onOutputStream(sink)
+                .withOptions(
+                    ProducerOptions.sign(
+                        SigningOptions.get()
+                            .overrideHashAlgorithm(HashAlgorithm.requireFromId(hashAlgorithm))
+                            .addDetachedSignature(keyProtector, key)
+                    ).setAsciiArmor(false)
+                        .overrideCompressionAlgorithm(CompressionAlgorithm.UNCOMPRESSED)
+                );
+        } catch (PGPException | IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void feedHeader(ByteBuffer header) {
+        feedData(header);
+    }
+
+    @Override
+    public void feedPayloadData(ByteBuffer data) {
+        feedData(data);
+    }
+
+    private void feedData(ByteBuffer data) {
+        try {
+            if (data.hasArray()) {
+                signingStream.write(data.array(), data.position(), data.remaining());
+            } else {
+                final byte[] buffer = new byte[data.remaining()];
+                data.get(buffer);
+                signingStream.write(buffer);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected Logger getLogger() {
+        return logger;
+    }
+
+    @Override
+    public void finish(Header<RpmSignatureTag> signature) {
+        try {
+            signingStream.close();
+            EncryptionResult result = signingStream.getResult();
+            PGPSignature pgpSignature = result.getDetachedSignatures().flatten().iterator().next();
+            byte[] value = pgpSignature.getEncoded();
+            switch (pgpSignature.getKeyAlgorithm()) {
+                // RSA
+                case PublicKeyAlgorithmTags.RSA_GENERAL: // 1
+                    getLogger().info("RSA: {}", value);
+                    signature.putBlob(RpmSignatureTag.PGP, value);
+                    break;
+
+                // DSA
+                case PublicKeyAlgorithmTags.DSA: // 17
+                case PublicKeyAlgorithmTags.EDDSA_LEGACY: // 22
```

Comment:
I think that must be a typo in the documentation and they mean EdDSA, not EcDSA. Note the capitalization, too, since ECDSA is written in all caps, but EdDSA isn't.

Reply:
I documented this mismatch :)